openedx/features/course_experience/tests/views/test_welcome_message.py

```diff
@@ -0,0 +1,90 @@
+"""
+Tests for course welcome messages.
+"""
+from unittest import skip
+
+from django.core.urlresolvers import reverse
+
+from student.models import CourseEnrollment
+from student.tests.factories import UserFactory
+from xmodule.modulestore import ModuleStoreEnum
+from xmodule.modulestore.django import modulestore
+from xmodule.modulestore.tests.django_utils import SharedModuleStoreTestCase
+from xmodule.modulestore.tests.factories import CourseFactory, ItemFactory
+from xmodule.html_module import CourseInfoModule
+
+TEST_PASSWORD = 'test'
+TEST_WELCOME_MESSAGE = '<h2>Welcome!</h2>';
+
+
+def welcome_message_url(course):
+    """
+    Returns the URL for the welcome message view.
+    """
+    return reverse(
+        'openedx.course_experience.welcome_message_fragment_view',
+        kwargs={
+            'course_id': unicode(course.id),
+        }
+    )
+
+
+def create_welcome_message(course, user, welcome_message=TEST_WELCOME_MESSAGE):
+    """
+    Creates a test welcome message for the specified course.
+    """
+    updates_usage_key = course.id.make_usage_key('course_info', 'updates')
+    course_updates = modulestore().create_item(
+        user.id,
+        updates_usage_key.course_key,
+        updates_usage_key.block_type,
+        block_id=updates_usage_key.block_id
+    )
+    course_updates.data = u'<ol><li><a href="test">Test Update</a></li></ol>'
+    course_updates.items.append({
```

Comment:
That was a great suggestion that turned out requiring a major rehaul of the way course updates were managed in the tests. New commit coming...